### ISSUE TITLE
mio-serial v3.0.0 not compatible with serialport v3.1.0

### ISSUE THREAD

**Issue by ntosta** [berkowski/mio-serial#14]
The serialport crate (https://docs.rs/crate/serialport/%5E3.0) which is listed as a dependency recently got updated to version 3.1.0 from version 3.0.0. In this change 3 new functions got added to the SerialPort trait, and as a result the Serial implementation in mio-serial won't compile:

```
...
Compiling serialport v3.1.0
Compiling mio-serial v3.0.0
error[E0046]: not all trait items implemented, missing: `bytes_to_read`, `bytes_to_write`, `clear`
   --> /home/nrt/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-serial-3.0.0/src/unix.rs:137:1
    |
137 | impl SerialPort for Serial {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `bytes_to_read`, `bytes_to_write`, `clear` in implementation
    |
    = note: `bytes_to_read` from trait: `fn(&Self) -> std::result::Result<u32, serialport::Error>`
    = note: `bytes_to_write` from trait: `fn(&Self) -> std::result::Result<u32, serialport::Error>`
    = note: `clear` from trait: `fn(&Self, serialport::ClearBuffer) -> std::result::Result<(), serialport::Error>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0046`.
error: Could not compile `mio-serial`.
```

Easiest fix would be to just update the cargo.toml to specify 3.0.0 until those functions can be implemented.

**Comment by berkowski** [comment]
@ntosta

Thanks.  I'll try to get a fix up to crates.io.  I have very slow internet at the moment.

**Comment by berkowski** [comment]
I was able to push a new release (3.0.1) that pins serialport-rs to 3.0.  I'll look into the 3.1 stuff next week.

**Comment by berkowski** [comment]
Closing this now that I've released versions for serialport-rs v3.1 and v3.2
